packages/core/src/browser/widgets/select-component.tsx

```diff
@@ -0,0 +1,301 @@
+            original: reset ? this.state.original : this.state.selected
+        });
+    }
+
+    protected renderDropdown(): React.ReactNode {
+        if (!this.state.dimensions) {
+            return;
+        }
+        if (this.mountedListeners.size === 0) {
+            // Only attach our listeners once we render our dropdown menu
+            this.attachListeners();
+            // We can now also calculate the optimal width
+            this.optimalWidth = this.getOptimalWidth();
+            this.optimalHeight = this.getOptimalHeight(Math.max(this.state.dimensions.width, this.optimalWidth));
+        }
+        const clientHeight = document.getElementById('theia-app-shell')!.getBoundingClientRect().height;
+        const invert = this.optimalHeight > clientHeight - this.state.dimensions.top;
+        const { options } = this.props;
+        const { selected } = this.state;
+        const description = selected !== undefined && options[selected].description;
+        const markdown = selected !== undefined && options[selected].markdown;
+        const items = options.map((item, i) => this.renderOption(i, item));
+        if (description) {
+            let descriptionNode: React.ReactNode | undefined;
+            const className = 'theia-select-component-description';
+            if (markdown) {
+                descriptionNode = <div key="description" className={className}
+                    dangerouslySetInnerHTML={{ __html: DOMPurify.sanitize(description) }} />; // eslint-disable-line react/no-danger
+            } else {
+                descriptionNode = <div key="description" className={className}>
+                    {description}
+                </div>;
+            }
+            if (invert) {
+                items.unshift(descriptionNode);
+            } else {
+                items.push(descriptionNode);
+            }
+        }
+        return <div key="dropdown" className="theia-select-component-dropdown" style={{
+            top: invert ? 'none' : this.state.dimensions.top,
+            bottom: invert ? clientHeight - this.state.dimensions.top + this.state.dimensions.parentHeight : 'none',
+            left: this.state.dimensions.left,
+            width: Math.max(this.state.dimensions.width, this.optimalWidth),
+            position: 'absolute'
+        }}>
+            {items}
+        </div>;
+    }
+
+    protected renderOption(index: number, option: SelectOption): React.ReactNode {
+        if (option.separator) {
+            return <div key={index} className="theia-select-component-separator" />;
+        }
+        const selected = this.state.selected;
+        return (
+            <div
+                key={index}
+                className={`theia-select-component-option${index === selected ? ' selected' : ''}`}
+                onMouseOver={() => {
+                    this.setState({
+                        selected: index
+                    });
+                }}
```

Comment:
It looks like this updates the the top-line selected item every time the user moves the mouse. In VSCode, the description changes `onMouseOver`, but the top line only changes on selection by click or keyboard action.